helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1360,6 +1369,8 @@ private boolean updateControllerState(NotificationContext changeContext, PauseSi
         String uid = UUID.randomUUID().toString().substring(0, 8);
         ClusterEvent event = new ClusterEvent(_clusterName, ClusterEventType.Resume,
             String.format("%s_%s", uid, Pipeline.Type.DEFAULT.name()));
+        event.addAttribute(AttributeName.EVENT_SESSION.name(),
+            changeContext.getManager().getSessionIdIfLead());
```

Comment:
This part is actually debatable? updateControllerState is used to pause or restart paused pipeline. Do it care about session? Can we lose this signal due to session check? How do we reason this one?

I am not very familiar with this logic. @jiajunwang , @lei-xia, @dasahcc

Reply:
@pkuwm , here is the thing. If say when pause signal znode is added, the under old session S0, the event is put into zkclient queue. Then a new session establish for S1, would the controller under S1 for this logic, would not pause of the old event. 

The question is that can you double check if the new session, controller would get another event of pausing due to controller cache refresh? This is to make sure pause will not be lost immediately followed by a controller change.